### ISSUE TITLE
Does Botan 1.10.17 support AES/GCM cipher

### ISSUE THREAD

**Issue by adnan-kamili** [randombit/botan#3268]
We have a legacy app where we use Botan 1.10.17 (can't use newer version of Botan as they don't support older Linux Distros). We use amalgamation build and wanted to use AES/GCM ciper. I tried following:

```
python3 configure.py --cc=clang --os=darwin --cpu=arm64 --no-autoload --gen-amalgamation --enable-modules=rsa,eme_pkcs,emsa3,emsa4,sha1,auto_rng,cbc,kdf2,asm_engine,pthreads,unix_procs,pbes2,sha2_64,des,
gcm,aead
```

But is gives a warning:

```
WARNING: Unknown enabled module aead
WARNING: Unknown enabled module gcm
```

**Comment by randombit** [comment]
Sorry no, GCM was added in 1.11.3

> can't use newer version of Botan as they don't support older Linux Distros

Just out of curiosity what distro/gcc are you using that doesn't have GCC 4.8?? Even RHEL7 has that  

Also FWIW a old distro should be fine as long as the compiler is updated.

**Comment by adnan-kamili** [comment]
We have to support Debian 7 with glibc v2.13 and gcc 4.7, and as per some previous discussion I remember Botan 2 doesn't work with this configuration.

**Comment by randombit** [comment]
You might be able to get away with compiling 1.11.3, or even a somewhat later version, with GCC 4.7. 1.11.0 was released before GCC 4.8, and the change notes claim it worked with GCC 4.7. 1.11.3 was released only a couple of weeks after GCC 4.8 was. I don't see in the notes where exactly we dropped GCC 4.7 support along the way, and I certainly don't remember since this was literally a decade ago, but you might get lucky. Especially if you're able to disable features you're not otherwise using. TLS was seeing very active work at that time and I imagine the first uses of GCC 4.8-only features was there.

If you decide to try this I really can't offer much in the way of support, 1.10 has been EOL for over 4 years and officially speaking 1.11 were development snapshots and never supported. But something to try if you're really stuck here.

... further digging ...

You may be in luck here, 30edf16ca6fabc22f mentions GCC 4.7, which means it may have worked as late as 1.11.27 (or even later, who knows).

Another thing that might be worth a shot is actually just compiling latest 2.19.3 release with GCC 4.7 and seeing what doesn't work. It may only require minor patching to compile, I have no idea. We test GCC 4.8 builds in CI still, and that's fine. And at least that release *is* supported so I'd be a bit more willing to spend time looking at compile errors and suggesting solutions if you go this route.

**Comment by randombit** [comment]
@adnan-kamili For 2.19.3, try the patch in #3273. The other thing you'll need to do for Debian 7 specifically is add the `configure.py` flag `--without-os-feature=getauxval`, since the glibc version in Debian 7 is too old.

**Comment by adnan-kamili** [comment]
Thanks a lot for your help, I appreciate it!

Will try giving 1.11.x a shot and see if I am able to get this working.

**Comment by adnan-kamili** [comment]
Does 1.10.17 support ChaCha20-Poly1305 and which variants? Original or original as well as ietf variant. We can use this instead of AES-GCM to get around this issue.

**Comment by randombit** [comment]
@adnan-kamili Sorry no, ChaCha20Poly1305 wasn't added until 1.11.12. The only AEAD included in 1.10 is EAX.

**Comment by randombit** [comment]
@adnan-kamili The patches to allow compiling 2.x using GCC 4.7 have landed, as I said I would suggest trying that first if this is something you need.
